### PR TITLE
Attempt to handle atp redirects from .fbx to .baked.fst

### PR DESCRIPTION
**Pull request by SamGondelman**
see the test plan from https://github.com/highfidelity/hifi/pull/14905 to make sure FSTs are still working
need to test with @Atlante45's Asset Server changes to make sure the redirect works

### CODE REVIEW

File: libraries/material-networking/src/material-networking/TextureCache.cpp [hunk 1]
```diff
@@ -630,11 +630,9 @@ void NetworkTexture::makeLocalRequest() {
 }
 
 bool NetworkTexture::handleFailedRequest(ResourceRequest::Result result) {
-    if (_currentlyLoadingResourceType != ResourceType::KTX
-        && result == ResourceRequest::Result::RedirectFail) {
-
+    if (_shouldFailOnRedirect && result == ResourceRequest::Result::RedirectFail) {
         auto newPath = _request->getRelativePathUrl();
-        if (newPath.fileName().endsWith(".ktx")) {
+        if (newPath.fileName().toLower().endsWith(".ktx")) {
```
Comment (SamGondelman) [inline libraries/material-networking/src/material-networking/TextureCache.cpp]: In order to fix skyboxes on the asset server, I think this should be changed to .texmeta.json, the line below should be ResourceType::META, and the logic where _shouldFailOnRedirect is set up above should be `_shouldFailOnRedirect = _currentlyLoadingResourceType != ResourceType::META;`

File: libraries/model-networking/src/model-networking/ModelCache.cpp [hunk 2]
```diff
@@ -300,47 +191,137 @@ void GeometryReader::run() {
...
+GeometryResource::GeometryResource(const GeometryResource& other) :
+    Resource(other),
+    Geometry(other),
+    _modelLoader(other._modelLoader),
+    _mappingPair(other._mappingPair),
+    _textureBaseURL(other._textureBaseURL),
+    _combineParts(other._combineParts),
+    _isCacheable(other._isCacheable)
+{
+    if (other._geometryResource) {
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/ModelCache.cpp]: If we're loading an FST and we've already seen the FST elsewhere but the other FBX download isn't finished, we need to do this request from scratch (same as NetworkTexture)

File: libraries/model-networking/src/model-networking/ModelCache.cpp [hunk 3]
```diff
@@ -300,47 +191,137 @@ void GeometryReader::run() {
...
+        _startedLoading = false;
+    }
+}
 
-    virtual void downloadFinished(const QByteArray& data) override;
+void GeometryResource::downloadFinished(const QByteArray& data) {
+    if (_activeUrl.fileName().toLower().endsWith(".fst")) {
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/ModelCache.cpp]: This is the switch on the redirected URL to see if we have an FST or FBX, the code below is the same as the two old downloadFinished methods

File: libraries/model-networking/src/model-networking/ModelCache.cpp [hunk 4]
```diff
@@ -300,47 +191,137 @@ void GeometryReader::run() {
...
+        }
+        QThreadPool::globalInstance()->start(new GeometryReader(_modelLoader, _self, _effectiveBaseURL, _mappingPair, data, _combineParts, _request->getWebMediaType()));
+    }
 }
 
-void GeometryDefinitionResource::downloadFinished(const QByteArray& data) {
-    if (_url != _effectiveBaseURL) {
-        _url = _effectiveBaseURL;
-        _textureBaseUrl = _effectiveBaseURL;
+bool GeometryResource::handleFailedRequest(ResourceRequest::Result result) {
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/ModelCache.cpp]: Based on NetworkTexture::handleFailedRequest

File: libraries/model-networking/src/model-networking/ModelCache.cpp [hunk 5]
```diff
@@ -388,26 +386,14 @@ ModelCache::ModelCache() {
...
-    } else {
-        resource = new GeometryDefinitionResource(_modelLoader, url);
-    }
-
-    return QSharedPointer<Resource>(resource, &Resource::deleter);
+    return QSharedPointer<Resource>(new GeometryResource(url, _modelLoader), &GeometryResource::deleter);
 }
 
 QSharedPointer<Resource> ModelCache::createResourceCopy(const QSharedPointer<Resource>& resource) {
-    if (resource->getURL().path().toLower().endsWith(".fst")) {
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/ModelCache.cpp]: This is where we used to do the extension check, before we loaded anything.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18697/HighFidelity-Beta-PR15374-4e795b6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18697/HighFidelity-Beta-Interface-PR15374-4e795b6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18697/HighFidelity-Beta-PR15374-4e795b6.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18697/HighFidelity-Quest-Beta-PR15374-4e795b6.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18697/HighFidelity-Beta-PR15374-4e795b6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18697/HighFidelity-Beta-Interface-PR15374-4e795b6.exe

**Pull request closed without merging**